### PR TITLE
Added v12 example and text

### PR DESCRIPTION
**Pull request by adrianhall**
Working with @kaerm on Azure SDK docs.

### CODE REVIEW

File: docs-ref-services/storage.md [hunk 1]
```diff
@@ -16,19 +15,79 @@ The Azure Storage libraries for Java provide classes for working with data in yo
...
+| Version 12 | Blob and Queue | [Blob](https://search.maven.org/artifact/com.azure/azure-storage-blob/)<br />[Queue](https://search.maven.org/artifact/com.azure/azure-storage-queue/) | [Blob](https://azuresdkdocs.blob.core.windows.net/$web/java/azure-storage-blob/12.0.0/index.html)<br/>[Queue](https://azuresdkdocs.blob.core.windows.net/$web/java/azure-storage-queue/12.0.0/index.html) | [Blob](https://github.com/Azure/azure-sdk-for-java/tree/azure-storage-blob_12.0.0/sdk/storage/azure-storage-blob)<br />[Queue](https://github.com/Azure/azure-sdk-for-java/tree/azure-storage-blob_12.0.0/sdk/storage/azure-storage-queue) |
+| Version 8 | Blob, Queue, File, and Table | [All services](https://mvnrepository.com/artifact/com.microsoft.azure/azure-storage) | [Version 8 reference](https://docs.microsoft.com/java/api/overview/azure/storage/client?view=azure-java-stable)<br/>[Quickstart](/azure/storage/blobs/storage-quickstart-blobs-java) | [All services](https://github.com/azure/azure-storage-java/tree/legacy-master) |
```
Comment (kaerm) [inline docs-ref-services/storage.md]: Quickstart link doesn't work

File: docs-ref-services/storage.md [hunk 2]
```diff
@@ -16,19 +15,79 @@ The Azure Storage libraries for Java provide classes for working with data in yo
...
+<dependency>
+  <groupId>com.azure</groupId>
+  <artifactId>azure-storage-queue</artifactId>
+  <version>12.0.0</version>
+</dependency>
+```
+
+For more information about adding a dependency in Java, see [Add a dependency](https://maven.apache.org/guides/getting-started/index.html#How_do_I_use_external_dependencies).
+
+## Example usage
```
Comment (kaerm) [inline docs-ref-services/storage.md]: should be ### Example usage

File: docs-ref-services/storage.md [hunk 3]
```diff
@@ -16,19 +15,79 @@ The Azure Storage libraries for Java provide classes for working with data in yo
...
+}
+
+/* Upload the file to the container */
+BlobClient blobClient = containerClient.getBlobClient("my-remote-file.jpg");
+blobClient.uploadFileFile("my-local-file.jpg");
+```
+
+For more examples, review the [Client Library README](https://github.com/Azure/azure-sdk-for-java/blob/azure-storage-blob_12.0.0/sdk/storage/azure-storage-blob/README.md).
+
+## Available packages
```
Comment (kaerm) [inline docs-ref-services/storage.md]: should be ### Available packages

File: docs-ref-services/storage.md [hunk 4]
```diff
@@ -16,19 +15,79 @@ The Azure Storage libraries for Java provide classes for working with data in yo
...
-| Version 8 | Blob, Queue, File, and Table | [All services](https://mvnrepository.com/artifact/com.microsoft.azure/azure-storage) | [Version 8 reference](https://docs.microsoft.com/java/api/overview/azure/storage/client?view=azure-java-stable) | [All services](https://github.com/azure/azure-storage-java/tree/legacy-master) |
+| Version 12 | Blob and Queue | [Blob](https://search.maven.org/artifact/com.azure/azure-storage-blob/)<br />[Queue](https://search.maven.org/artifact/com.azure/azure-storage-queue/) | [Blob](https://azuresdkdocs.blob.core.windows.net/$web/java/azure-storage-blob/12.0.0/index.html)<br/>[Queue](https://azuresdkdocs.blob.core.windows.net/$web/java/azure-storage-queue/12.0.0/index.html) | [Blob](https://github.com/Azure/azure-sdk-for-java/tree/azure-storage-blob_12.0.0/sdk/storage/azure-storage-blob)<br />[Queue](https://github.com/Azure/azure-sdk-for-java/tree/azure-storage-blob_12.0.0/sdk/storage/azure-storage-queue) |
```
Comment (kaerm) [inline docs-ref-services/storage.md]: I'd add links to Quickstarts on docs - they are updated

File: docs-ref-services/storage.md (hunk 1 above)
Comment (adrianhall) [reply docs-ref-services/storage.md]: Link is relative to docs, so it should work once the build is done.  I didn't change it - just moved it.

File: docs-ref-services/storage.md (hunk 4 above)
Comment (adrianhall) [reply docs-ref-services/storage.md]: Not sure what you mean there.  The heading is "Source, Readme, Examples" - and all those are in GitHub.

File: docs-ref-services/storage.md (hunk 1 above)
Comment (kaerm) [reply docs-ref-services/storage.md]: Okay - but that link points to v12 not v8 :). v8 would use /azure/storage/blobs/storage-quickstart-blobs-java-legacy

File: docs-ref-services/storage.md (hunk 4 above)
Comment (kaerm) [reply docs-ref-services/storage.md]: Right, I did not think about the layout. maybe another column? I just think that would be a good pointer.

**Comment by adrianhall** [comment]
#sign-off

File: docs-ref-services/storage.md [hunk 5]
```diff
@@ -16,19 +15,79 @@ The Azure Storage libraries for Java provide classes for working with data in yo
...
+| Version 8 | Blob, Queue, File, and Table | [All services](https://mvnrepository.com/artifact/com.microsoft.azure/azure-storage) | [Version 8 reference](https://docs.microsoft.com/java/api/overview/azure/storage/client?view=azure-java-stable) | [All services](https://github.com/azure/azure-storage-java/tree/legacy-master) ([Quickstart](/azure/storage/blobs/storage-quickstart-blobs-java-legacy)) |
+
+### Accessing other storage account services with v12 libraries
 
-To use the client library in your project, add a dependency to your Maven `pom.xml` file. To learn how to add the dependency, visit the source repository listed in the table above for the client library you wish to use. For more information about adding a dependency in Java, see [Add a dependency](https://maven.apache.org/guides/getting-started/index.html#How_do_I_use_external_dependencies).
+Currently, the following libraries are in public preview:
 
-For more information about using the Azure Storage client library for Java, see the following articles:
+- File shares
+- File (Data lake)
```
Comment (kaerm) [inline docs-ref-services/storage.md]: One more comment from the storage team - let's just call data lake Data Lake without File

File: docs-ref-services/storage.md [hunk 6]
```diff
@@ -16,19 +15,79 @@ The Azure Storage libraries for Java provide classes for working with data in yo
 
 ## Client library for data access
 
-Use the Azure Storage client library for Java to work with data in your storage account. The following table describes the recommended versions of the storage client library for Java.
+Use the Azure Storage client library for Java to work with data in your storage account:
+
+- Read and write objects and files from [Azure blob storage](/azure/storage/blobs/storage-blobs-introduction)
```
Comment (mhopkins-msft) [inline docs-ref-services/storage.md]: Azure Blob is a service name, so we capitalize the word Blob in the link: \[Azure Blob storage\]

File: docs-ref-services/storage.md [hunk 7]
```diff
@@ -16,19 +15,79 @@ The Azure Storage libraries for Java provide classes for working with data in yo
...
+- Send and receive messages between cloud-connected applications with [Azure queue storage](/azure/storage/queues/storage-queues-introduction)
```
Comment (mhopkins-msft) [inline docs-ref-services/storage.md]: Azure Queue is a service name, so we capitalize the word Queue in the link: \[Azure Queue storage\]

Comment (adrianhall) [reply docs-ref-services/storage.md]: ack

**Pull request merged**